js/components/directionControl.js

```diff
@@ -0,0 +1,39 @@
+var React = require('react');
+var classnames = require('classnames');
+
+var DirectionControl = React.createClass({
+  getInitialState: function() {
+    return {
+      isTouched: false,
+    };
+  },
+
+  handleEvent: function (ev) {
+    var rotate = ev.type == 'mousedown' || ev.type == 'touchstart';
+    this.props.handleDirection(rotate, this.props.dir);
+    
+    this.setState({
+      isTouched: rotate
+    });
+  },
+
+  render: function () {
+    var directionClass = 'oo-vr-icon--move--' + this.props.dir
+      , touchedDirectionClass = this.state.isTouched && 'oo-vr-icon--move--' + this.props.dir + '--touched';
```

Comment:
You are using a bitwise operation && on a boolean and a string.
This will going to do type coercion from string to a boolean type, correct?

Reply:
I checked that this is evaluated this way:
if isTouched is true, then touchedDirectionClass will be  'oo-vr-icon--move--' + this.props.dir + '--touched'
if isTouched is false, then touchedDirectionClass will be false.

Reply:
yes, that is true, but touchedDirectionClass is used in classnames, so if touchedDirectionClass is false classnames ignores it.